### PR TITLE
withinPolygon: 3 point minimum and boundary testing.

### PR DESCRIPTION
**Pull request by dplewis**
Mongo has a 3 point minimum for $polygon and throws an error.

Also withinPolygon and withinGeoBox are inclusive

They return within and on bounds. Updated test for inclusion

### CODE REVIEW

File: src/Adapters/Storage/Mongo/MongoTransform.js [hunk 1]
```diff
@@ -623,6 +623,12 @@ function transformConstraint(constraint, inArray) {
       if (!(polygon instanceof Array)) {
         throw new Parse.Error(Parse.Error.INVALID_JSON, 'bad $geoWithin value');
       }
+      if (polygon.length < 3) {
+        throw new Parse.Error(
+          Parse.Error.INVALID_JSON,
+          'Polygon must have at least 3 points'
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: can you be more explicit on that: ex:

```
bad $geoWithin value; $polygon should contain at least 3 GeoPoints
```

File: src/Adapters/Storage/Mongo/MongoTransform.js [hunk 2]
```diff
@@ -623,6 +623,12 @@ function transformConstraint(constraint, inArray) {
       if (!(polygon instanceof Array)) {
         throw new Parse.Error(Parse.Error.INVALID_JSON, 'bad $geoWithin value');
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoTransform.js]: can we be more specific here with:

```
bad $geoWithin value; $polygon should contain at least 3 GeoPoints
```

File: src/Adapters/Storage/Postgres/PostgresStorageAdapter.js [hunk 3]
```diff
@@ -350,6 +350,12 @@ const buildWhereClause = ({ schema, query, index }) => {
       if (!(polygon instanceof Array)) {
         throw new Parse.Error(Parse.Error.INVALID_JSON, 'bad $geoWithin value');
       }
+      if (polygon.length < 3) {
+        throw new Parse.Error(
+          Parse.Error.INVALID_JSON,
+          'Polygon must have at least 3 points'
```
Comment (flovilmart) [inline src/Adapters/Storage/Postgres/PostgresStorageAdapter.js]: same here :)

**Comment by dplewis** [comment]
I've added an additional test for open vs closed polygons. They both are supported

**Comment by dplewis** [comment]
Thanks

**Pull request merged**